[2025-07-11T15:58:21.031+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-11T15:58:21.052+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: employee_ingestion_dag.send_alerts manual__2025-07-11T15:58:13+00:00 [queued]>
[2025-07-11T15:58:21.061+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: employee_ingestion_dag.send_alerts manual__2025-07-11T15:58:13+00:00 [queued]>
[2025-07-11T15:58:21.062+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-11T15:58:21.083+0000] {taskinstance.py:2890} INFO - Executing <Task(_PythonDecoratedOperator): send_alerts> on 2025-07-11 15:58:13+00:00
[2025-07-11T15:58:21.097+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=589) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-07-11T15:58:21.102+0000] {standard_task_runner.py:72} INFO - Started process 595 to run task
[2025-07-11T15:58:21.102+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'employee_ingestion_dag', 'send_alerts', 'manual__2025-07-11T15:58:13+00:00', '--job-id', '92', '--raw', '--subdir', 'DAGS_FOLDER/ingestion_pipeline.py', '--cfg-path', '/tmp/tmpbmeu1dgz']
[2025-07-11T15:58:21.103+0000] {standard_task_runner.py:105} INFO - Job 92: Subtask send_alerts
[2025-07-11T15:58:21.160+0000] {task_command.py:467} INFO - Running <TaskInstance: employee_ingestion_dag.send_alerts manual__2025-07-11T15:58:13+00:00 [running]> on host 5a07e0ddeb4c
[2025-07-11T15:58:21.234+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='employee_ingestion_dag' AIRFLOW_CTX_TASK_ID='send_alerts' AIRFLOW_CTX_EXECUTION_DATE='2025-07-11T15:58:13+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-11T15:58:13+00:00'
[2025-07-11T15:58:21.234+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-11T15:58:26.303+0000] {ingestion_pipeline.py:197} WARNING - TEAMS_WEBHOOK_URL not configured, skipping Teams alert
[2025-07-11T15:58:26.324+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/great_expectations/data_context/data_context/serializable_data_context.py:225: UserWarning: Warning. An existing `great_expectations.yml` was found here: /opt/airflow/gx.
    - No action was taken.
  warnings.warn(message)

[2025-07-11T15:58:26.325+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/great_expectations/data_context/data_context/serializable_data_context.py:233: UserWarning: Warning. An existing `config_variables.yml` was found here: /opt/airflow/gx/uncommitted.
    - No action was taken.
  warnings.warn(message)

[2025-07-11T15:58:26.345+0000] {file_data_context.py:222} INFO - FileDataContext loading fluent config
[2025-07-11T15:58:26.352+0000] {config.py:187} INFO - Loading 'datasources' ->
[{'assets': [...], 'name': 'employee_datasource', 'type': 'pandas'}]
[2025-07-11T15:58:26.353+0000] {fluent_base_model.py:276} INFO - _PandasDataAsset.dict() - missing `config_provider`, skipping config substitution
[2025-07-11T15:58:26.453+0000] {abstract_data_context.py:5516} INFO - Loaded 'employee_datasource' from fluent config
[2025-07-11T15:58:26.833+0000] {ingestion_pipeline.py:212} INFO - Data docs built successfully. Report URL: http://localhost:8085/
[2025-07-11T15:58:26.836+0000] {ingestion_pipeline.py:249} INFO - 
            ðŸš¨ DATA VALIDATION ALERT ðŸš¨
            File: employee_data_with_errors_chunk_2.csv
            Total Rows: 100
            Valid Rows: 89
            Invalid Rows: 11
            Criticality: MEDIUM
            Issues: Age: expect_column_values_to_not_be_null failed (11)
            Report: http://localhost:8085/
            
[2025-07-11T15:58:26.837+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-07-11T15:58:26.850+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-11T15:58:26.851+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=employee_ingestion_dag, task_id=send_alerts, run_id=manual__2025-07-11T15:58:13+00:00, execution_date=20250711T155813, start_date=20250711T155821, end_date=20250711T155826
[2025-07-11T15:58:26.945+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-11T15:58:26.965+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-11T15:58:26.967+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
